### PR TITLE
[Fix #18] Get it ready for republishing on hex.pm

### LINKED ISSUE #18: Switch build tools to erlang.mk and republish to hex.pm

### PR DESCRIPTION
**Pull request by elbrujohalcon**
## I also merged the newest changes from upstream

### CODE REVIEW

File: src/aleppo.erl [hunk 1]
```diff
@@ -403,26 +411,51 @@ mark_keywords([{'-', DashAttrs} = Dash,
...
+
+reverse_and_normalize_token_locations(RevTokens) ->
+    reverse_and_normalize_token_locations_helper(RevTokens, []).
+
+reverse_and_normalize_token_locations_helper([], Acc) ->
+    Acc;
+reverse_and_normalize_token_locations_helper([{Type, MaybeLocation} | Rest], Acc) when is_tuple(MaybeLocation) orelse
+                                                                                       is_list(MaybeLocation) ->
+    reverse_and_normalize_token_locations_helper(Rest, [{Type, location(MaybeLocation)}|Acc]);
+reverse_and_normalize_token_locations_helper([{Type, MaybeLocation, Extra} | Rest], Acc) when is_tuple(MaybeLocation) orelse
```
Comment (gadgetci) [inline src/aleppo.erl]: According to **Elvis**:
> Line 457 is too long: reverse_and_normalize_token_locations_helper([{Type, MaybeLocation, Extra} | Rest], Acc) when is_tuple(MaybeLocation) orelse.

File: src/aleppo.erl [hunk 2]
```diff
@@ -9,10 +9,13 @@
...
 
 process_file(FileName) ->
+    process_file(FileName, []).
+
+process_file(FileName, Options) ->
     ModuleName = list_to_atom(filename:rootname(filename:basename(FileName))),
     case scan_file(FileName) of
         {ok, Tokens} ->
-            process_tokens(Tokens, [{file, FileName}, {module, ModuleName}]);
+            process_tokens(Tokens, [{file, FileName}, {module, ModuleName}|Options]);
```
Comment (gadgetci) [inline src/aleppo.erl]: According to **Elvis**:
> Line 18 is too long:             process_tokens(Tokens, [{file, FileName}, {module, ModuleName}|Options]);.

File: src/aleppo.erl [hunk 3]
```diff
@@ -403,26 +411,51 @@ mark_keywords([{'-', DashAttrs} = Dash,
...
+reverse_and_normalize_token_locations(RevTokens) ->
+    reverse_and_normalize_token_locations_helper(RevTokens, []).
+
+reverse_and_normalize_token_locations_helper([], Acc) ->
+    Acc;
+reverse_and_normalize_token_locations_helper([{Type, MaybeLocation} | Rest], Acc) when is_tuple(MaybeLocation) orelse
+                                                                                       is_list(MaybeLocation) ->
+    reverse_and_normalize_token_locations_helper(Rest, [{Type, location(MaybeLocation)}|Acc]);
+reverse_and_normalize_token_locations_helper([{Type, MaybeLocation, Extra} | Rest], Acc) when is_tuple(MaybeLocation) orelse
+                                                                                              is_list(MaybeLocation) ->
```
Comment (gadgetci) [inline src/aleppo.erl]: According to **Elvis**:
> Line 458 is too long:                                                                                               is_list(MaybeLocation) ->.

File: src/aleppo.erl [hunk 4]
```diff
@@ -403,26 +411,51 @@ mark_keywords([{'-', DashAttrs} = Dash,
...
+    {Line, Column}.
+
+reverse_and_normalize_token_locations(RevTokens) ->
+    reverse_and_normalize_token_locations_helper(RevTokens, []).
+
+reverse_and_normalize_token_locations_helper([], Acc) ->
+    Acc;
+reverse_and_normalize_token_locations_helper([{Type, MaybeLocation} | Rest], Acc) when is_tuple(MaybeLocation) orelse
+                                                                                       is_list(MaybeLocation) ->
+    reverse_and_normalize_token_locations_helper(Rest, [{Type, location(MaybeLocation)}|Acc]);
```
Comment (gadgetci) [inline src/aleppo.erl]: According to **Elvis**:
> Line 456 is too long:     reverse_and_normalize_token_locations_helper(Rest, [{Type, location(MaybeLocation)}|Acc]);.

File: src/aleppo.erl [hunk 5]
```diff
@@ -403,26 +411,51 @@ mark_keywords([{'-', DashAttrs} = Dash,
...
+    Column = proplists:get_value(column, Attrs),
+    {Line, Column}.
+
+reverse_and_normalize_token_locations(RevTokens) ->
+    reverse_and_normalize_token_locations_helper(RevTokens, []).
+
+reverse_and_normalize_token_locations_helper([], Acc) ->
+    Acc;
+reverse_and_normalize_token_locations_helper([{Type, MaybeLocation} | Rest], Acc) when is_tuple(MaybeLocation) orelse
+                                                                                       is_list(MaybeLocation) ->
```
Comment (gadgetci) [inline src/aleppo.erl]: According to **Elvis**:
> Line 455 is too long:                                                                                        is_list(MaybeLocation) ->.

File: src/aleppo.erl [hunk 6]
```diff
@@ -1,6 +1,6 @@
 % Aleppo: ALternative Erlang Pre-ProcessOr
 -module(aleppo).
--export([process_file/1, process_tokens/1, process_tokens/2, scan_file/1]).
+-export([process_file/1, process_file/2, process_tokens/1, process_tokens/2, scan_file/1]).
```
Comment (gadgetci) [inline src/aleppo.erl]: According to **Elvis**:
> Line 3 is too long: -export([process_file/1, process_file/2, process_tokens/1, process_tokens/2, scan_file/1])..

File: src/aleppo.erl [hunk 7]
```diff
@@ -403,26 +411,51 @@ mark_keywords([{'-', DashAttrs} = Dash,
...
+    reverse_and_normalize_token_locations_helper(RevTokens, []).
+
+reverse_and_normalize_token_locations_helper([], Acc) ->
+    Acc;
+reverse_and_normalize_token_locations_helper([{Type, MaybeLocation} | Rest], Acc) when is_tuple(MaybeLocation) orelse
+                                                                                       is_list(MaybeLocation) ->
+    reverse_and_normalize_token_locations_helper(Rest, [{Type, location(MaybeLocation)}|Acc]);
+reverse_and_normalize_token_locations_helper([{Type, MaybeLocation, Extra} | Rest], Acc) when is_tuple(MaybeLocation) orelse
+                                                                                              is_list(MaybeLocation) ->
+    reverse_and_normalize_token_locations_helper(Rest, [{Type, location(MaybeLocation), Extra}|Acc]);
```
Comment (gadgetci) [inline src/aleppo.erl]: According to **Elvis**:
> Line 459 is too long:     reverse_and_normalize_token_locations_helper(Rest, [{Type, location(MaybeLocation), Extra}|Acc]);.

File: src/aleppo.erl (hunk 1 above)
Comment (gadgetci) [inline src/aleppo.erl]: According to **Elvis**:
> Line 457 is too long: reverse_and_normalize_token_locations_helper([{Type, MaybeLocation, Extra} | Rest], Acc) when is_tuple(MaybeLocation) orelse.

File: src/aleppo.erl (hunk 7 above)
Comment (gadgetci) [inline src/aleppo.erl]: According to **Elvis**:
> Line 459 is too long:     reverse_and_normalize_token_locations_helper(Rest, [{Type, location(MaybeLocation), Extra}|Acc]);.

File: src/aleppo.erl (hunk 4 above)
Comment (gadgetci) [inline src/aleppo.erl]: According to **Elvis**:
> Line 456 is too long:     reverse_and_normalize_token_locations_helper(Rest, [{Type, location(MaybeLocation)}|Acc]);.

File: src/aleppo.erl (hunk 2 above)
Comment (gadgetci) [inline src/aleppo.erl]: According to **Elvis**:
> Line 18 is too long:             process_tokens(Tokens, [{file, FileName}, {module, ModuleName}|Options]);.

File: src/aleppo.erl (hunk 6 above)
Comment (gadgetci) [inline src/aleppo.erl]: According to **Elvis**:
> Line 3 is too long: -export([process_file/1, process_file/2, process_tokens/1, process_tokens/2, scan_file/1])..

File: src/aleppo.erl (hunk 3 above)
Comment (gadgetci) [inline src/aleppo.erl]: According to **Elvis**:
> Line 458 is too long:                                                                                               is_list(MaybeLocation) ->.

File: src/aleppo.erl [hunk 8]
```diff
@@ -403,26 +411,51 @@ mark_keywords([{'-', DashAttrs} = Dash,
...
+    Line = proplists:get_value(line, Attrs),
+    Column = proplists:get_value(column, Attrs),
+    {Line, Column}.
+
+reverse_and_normalize_token_locations(RevTokens) ->
+    reverse_and_normalize_token_locations_helper(RevTokens, []).
+
+reverse_and_normalize_token_locations_helper([], Acc) ->
+    Acc;
+reverse_and_normalize_token_locations_helper([{Type, MaybeLocation} | Rest], Acc) when is_tuple(MaybeLocation) orelse
```
Comment (gadgetci) [inline src/aleppo.erl]: According to **Elvis**:
> Line 454 is too long: reverse_and_normalize_token_locations_helper([{Type, MaybeLocation} | Rest], Acc) when is_tuple(MaybeLocation) orelse.

File: src/aleppo.erl (hunk 5 above)
Comment (gadgetci) [inline src/aleppo.erl]: According to **Elvis**:
> Line 455 is too long:                                                                                        is_list(MaybeLocation) ->.

File: src/aleppo.erl (hunk 8 above)
Comment (gadgetci) [inline src/aleppo.erl]: According to **Elvis**:
> Line 454 is too long: reverse_and_normalize_token_locations_helper([{Type, MaybeLocation} | Rest], Acc) when is_tuple(MaybeLocation) orelse.

**Comment by elbrujohalcon** [comment]
@jfacorro I bumped the version, but I also merged from upstream where many changes (including rebar compilation) were added.
Please give this a thorough review and, if possible, let us know if this doesn't break katana or elvis and if you think, in that case, that we should actually fix katana and/or elvis after merging this PR.

**Pull request merged**